Move Software Model Check task to base plugin

ThadHouse opened the pull request:
Previously, a check task was only added if the testing base was added, causing confusion to users who wanted to add their own check tasks which are not directly unit tests.

Fixes #10851 

### Context
The check tasks shouldn't be dependent on the testing base being added, theyre not in both the new native setup or in Java code builds.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

ThadHouse commented:
I fixed the build failure related to this PR, so this should be good to go now.

rieske commented:
Thank you for the contribution.
Can you also provide integration tests to verify the behavior that this change fixes?

ThadHouse commented:
@rieske Finally got around to updating this PR, fixed the merge conflicts, fixed the existing test that was failing, and added a new integration test. This should be good to go now.

ThadHouse commented:
This pull request has been ready to go for months. It's a simple change, so there shouldn't be any major issues merging it. Especially with tests

rieske commented:
Thanks, @ThadHouse, and apologies for the delay!
